src/components/select-field/BaseSelectField.js

```diff
@@ -349,7 +360,7 @@ class BaseSelectField extends React.Component<Props, State> {
     selectMultiOption = (index: number) => {
         const { defaultValue, options, selectedValues } = this.props;
         const hasDefaultValue = defaultValue != null; // Checks if not undefined or null
-        const item = options[index];
+        const item = this.getItemFromFilteredOptions(index);
```

Comment:
Same goes for this line. Single / Multi Select can both be filtered on so this functionality holds true for this case.